extensions/email/src/main/java/org/mule/extension/email/api/retriever/AbstractRetrieverProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.api.retriever;
+
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.display.Password;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+
+/**
+ * Generic contract for all email retriever {@link ConfigurationProvider}s.
+ *
+ * @since 4.0
+ */
+public abstract class AbstractRetrieverProvider
```

Comment:
There is a bug that we need to fix in order to do that. I'll add a todo once the jira is created